### PR TITLE
Adding automatic variants

### PR DESCRIPTION
**Pull request by karanvirsb**
This is to add to pull request #188. This adds a checkbox when adding a new colour to automatically generate variants.

### CODE REVIEW

**Comment by alexdanilowicz** [comment]
@karanvirsb are you saying you want to merge this into Paul's branch: https://github.com/Mirrorful/mirrorful/pull/188 ? Or want to merge this into main?

**Comment by karanvirsb** [comment]
> ?

@alexdanilowicz It can be merged into the main if it seems good. I was just suggesting that it can also be done like this to @prichter.

File: packages/mirrorful/editor/src/components/ColorPalette/EditColorModal.tsx [hunk 1]
```diff
@@ -143,6 +148,17 @@ export function EditColorModal({
...
               />
             </FormControl>
+            {!initialColorData || base !== initialColorData.baseColor ? (
+              <FormControl>
+                <Checkbox
+                  checked={shouldGenerateVariants}
+                  onChange={() => setShouldGenerateVariants((prev) => !prev)}
+                  defaultChecked={shouldGenerateVariants}
+                >
+                  Automatically Generate Variants
```
Comment (alexdanilowicz) [inline packages/mirrorful/editor/src/components/ColorPalette/EditColorModal.tsx:158]:
```suggestion
                  Automatically generate variants
```

**Comment by alexdanilowicz** [comment]
Oh nice! Love it @karanvirsb. Thanks for clarifying. I just QA-ed and watched your Loom. Really nice. Merging.

**Pull request merged**